src/sqlfluff/core/parser/grammar/conditional.py

```diff
@@ -0,0 +1,67 @@
+"""Conditional Grammar."""
+
+from sqlfluff.core.parser.segments import Indent
+from sqlfluff.core.parser.match_result import MatchResult
+from sqlfluff.core.parser.match_wrapper import match_wrapper
+
+from sqlfluff.core.parser.grammar.base import (
+    BaseGrammar,
+)
+
+
+class Conditional(BaseGrammar):
+    """A grammar which is conditional on the parse context.
+
+    Args:
+        enforce_whitespace_preceeding (:obj:`bool`): Should the GreedyUntil
+            match only match the content if it's preceded by whitespace?
+            (defaults to False). This is useful for some keywords which may
+            have false alarms on some array accessors.
+
+    """
+
+    def __init__(self, *args, config_type: str = "", **kwargs):
+        if not all(issubclass(arg, Indent) for arg in args):
+            raise ValueError(
+                "Conditional is only designed to work with Indent segments."
+            )
+        if len(args) != 1:
+            raise ValueError(
+                "Conditional is only designed to work with a single element."
+            )
+        if not config_type:
+            raise ValueError("Conditional config_type must be set.")
+        if not kwargs:
+            raise ValueError("Conditional requires rules to be set.")

```

Comment:
What's the connection between `kwargs` and `rules`? Are we assuming that `kwargs` will contain `rules` as a key if it's not empty?

Reply:
correct. I might just rename `kwargs` to `rules` for clarity